src/components/home/__tests__/home.test.js

```diff
@@ -1,10 +1,431 @@
+        medium:
+          "https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/53305470/3/?bust=1634610256&width=300",
+        large:
+          "https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/53305470/3/?bust=1634610256&width=600",
+        full: "https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/53305470/3/?bust=1634610256",
+      },
+      {
+        small:
+          "https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/53305470/4/?bust=1634610257&width=100",
+        medium:
+          "https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/53305470/4/?bust=1634610257&width=300",
+        large:
+          "https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/53305470/4/?bust=1634610257&width=600",
+        full: "https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/53305470/4/?bust=1634610257",
+      },
+    ],
+    primary_photo_cropped: {
+      small:
+        "https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/53305470/1/?bust=1634610254&width=300",
+      medium:
+        "https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/53305470/1/?bust=1634610254&width=450",
+      large:
+        "https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/53305470/1/?bust=1634610254&width=600",
+      full: "https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/53305470/1/?bust=1634610254",
+    },
+    videos: [],
+    status: "adoptable",
+    status_changed_at: "2021-10-19T02:25:13+0000",
+    published_at: "2021-10-19T02:25:13+0000",
+    distance: null,
+    contact: {
+      email: "monica@angelsforpaws.org",
+      phone: "(949) 637-4326",
+      address: {
+        address1: null,
+        address2: null,
+        city: "Mesa",
+        state: "AZ",
+        postcode: "85205",
+        country: "US",
+      },
+    },
+    _links: {
+      self: {
+        href: "/v2/animals/53305470",
+      },
+      type: {
+        href: "/v2/types/dog",
+      },
+      organization: {
+        href: "/v2/organizations/az715",
+      },
+    },
+  },
+];
+
+jest.mock("../../../utils/homePageFetcher");
+
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+}));
+
+// These tests test pet list that have been returned by fetch
+describe("Testing pet list with mock list", () => {
+  beforeEach(async () => {
+    fetcher.mockReturnValue(petList);
+  });
+
+  test("list of Pets renders correctly", async () => {
+    const { findByText } = customRender(<Home />);
+    const namePet1 = await findByText(/Harley/i);
+    expect(namePet1).toBeInTheDocument();
+
+    const namePet2 = await findByText(/ISRAEL/i);
+    expect(namePet2).toBeInTheDocument();
+
+    const namePet3 = await findByText(/Andy & Mandy/i);
+    expect(namePet3).toBeInTheDocument();
+  });
+
+  test("calls SWR mutate when refresh button is clicked", async () => {
+    const mutate = jest.fn();
+    const { findByTestId } = customRender(<Home onClick={mutate()} />);
+    const button = await findByTestId("btn-refresh");
+    fireEvent.click(button);
```

Comment:
switch to using `userEvent` since it provides several methods that resemble the user interactions more closely